### ISSUE TITLE
UK population estimates for neighbourhood/borough

### ISSUE THREAD

**Issue by missinglink** [whosonfirst-data/whosonfirst-data#1887]
Heya,

We noticed that London Districts (wof:neighbourhood) don't have population estimates.
eg. [Putney](https://spelunker.whosonfirst.org/id/85789965/), [Clapham](https://spelunker.whosonfirst.org/id/85861663/)

Looking up one level in the hierarchy at London Boroughs (wof:borough) they also don't have population info:
eg. [Wandsworth](https://spelunker.whosonfirst.org/id/1158857251/), [Hackney](https://spelunker.whosonfirst.org/id/1158857273/)

I'd like to fix these, particularly in London but also for all of the UK for these two placetypes if a data source can be found.
I had a look around the UK govt stats websites but wasn't able to find anything granular enough to cover the neighbourhood records.

@vicchi do you possibly know a source of fairly recent population estimates for these places?

Since a recent change in Pelias we're now seeing the Putney train station returned above the neighbourhood because it's being considered more 'important', adding population info will help us to maintain a better balance :)

**Comment by justinelliotmeyers** [comment]
https://www.ons.gov.uk/peoplepopulationandcommunity/populationandmigration/populationestimates/bulletins/annualsmallareapopulationestimates/mid2018/relateddata

**Comment by missinglink** [comment]
I saw those, theres so many that sounds similar it's very confusing.

[This one](https://www.ons.gov.uk/peoplepopulationandcommunity/populationandmigration/populationestimates/datasets/lowersuperoutputareamidyearpopulationestimatesnationalstatistics) has stats for `Wandsworth` and then it drills down into 'Wards', it unfortunately only has the lower levels defined by their LSOA code.

AFAIK we don't have any mapping between LSOA and WOF, for instance I see there is a LSOA for 'East Putney' and 'West Putney' here http://www.uklocalarea.com/browse.php/Wandsworth/

Any tips about how to either find a file which has corresponding population counts, or building a `LSOA -> WOF` mapping would be super helpful :)

**Comment by missinglink** [comment]
<img width="497" alt="Screenshot 2020-09-01 at 16 19 40" src="https://user-images.githubusercontent.com/738069/91863026-f2a9e680-ec6e-11ea-94fe-369ca8356db9.png">

**Comment by justinelliotmeyers** [comment]
OAs are at a lower level than the LSOAs. I'm just trying to help you get the lowest level census poly that could in theory help you build up a population dataset for London. There are a couple ways we could back apply an even distribution based on UPRN and how they fall in these. Its probably an hour or two worth of work.

**Comment by missinglink** [comment]
A few WOF records seem to have a corresponding GSS code such as the record for [Wandsworth](https://data.whosonfirst.org/115/885/725/1/1158857251.geojson) but it's unfortunately not very common?

<img width="313" alt="Screenshot 2020-09-01 at 16 32 13" src="https://user-images.githubusercontent.com/738069/91864702-d27b2700-ec70-11ea-8f83-048937ea5d22.png">

**Comment by vicchi** [comment]
@missinglink This is all very timely and fits into the (still in progress) work I'm going with WOF UK admin boundaries ... basically the key to all of this is adding in to `wof:concordances` the GSS code ... with the caveat that not all of the ONS data will make it into WOF.

So you can map ONS census geographies (OAs, MSOAs and LSOAs) to (electoral) wards, such as this mapping for LSOAs - https://geoportal.statistics.gov.uk/search?collection=Dataset&sort=name&tags=all(LUP_LSOA_WD) and then wards to LADs (local authority districts) at which point you've jumped from census geography to admin geography, via electoral geography. All of these have GSS codes registered.

And ... when I'm finished, all of the UK admin records in WOF will have GSS code concordances.

There's a lot of work to do and also there's not always a direct mapping between all of these entities, though some of the ONS "best fit" mappings do help.

Also also, `gbr-datalondon:GSS_CODE` is a bit misleading, there's a few references to GSS codes for the UK in WOF but under different WOF properties, which also need to be cleaned up and (probably) pointed back to ONS as the authoritative data source as ONS "owns" and maintains the codes - https://geoportal.statistics.gov.uk/datasets/register-of-geographic-codes-december-2019-for-the-united-kingdom

So basically, yes we can, but not right now

**Comment by missinglink** [comment]
Aw shucks, why does this all have to be so complicated 😆 

I think in the interim I might just spatially intersect the LSOA geometries (if I can find them) with the WOF geometries and distribute the head counts based on overlap percentage.

It's not super accurate but it'll fix our issue while we wait for the proper fix to land.

**Comment by missinglink** [comment]
hmmm... or not, that doesn't seem very accurate...

<img width="835" alt="Screenshot 2020-09-01 at 16 55 30" src="https://user-images.githubusercontent.com/738069/91867637-276c6c80-ec74-11ea-9a9b-8d526b174dc4.png">
vs. Wikipedia

<img width="299" alt="Screenshot 2020-09-01 at 16 56 39" src="https://user-images.githubusercontent.com/738069/91867648-2afff380-ec74-11ea-9987-0597ffa1d2b9.png">

**Comment by justinelliotmeyers** [comment]
map all uprns to OAs then back apply based on even distribution.

**Comment by justinelliotmeyers** [comment]
will upload UPRNs when done doing join and analysis.

These are the OAs with 2018 estimates joined:
https://drive.google.com/file/d/1Sc2NZjWXPuKRVtH65kPjQKjiDKGRPAtO/view?usp=sharing

**Comment by justinelliotmeyers** [comment]
@missinglink were these useful?
